### DISCUSSION TITLE
Ability to pass {{domain}} to env variables

### DISCUSSION THREAD

**Discussion by w7tf** [coollabsio/coolify#2479] [Feature Requests]
In the preview deployment tab a number of URL templates can be set. I'd like to have the ability to pass down the domain as an environment variable. 

![image](https://github.com/coollabsio/coolify/assets/53350046/dc5cbfc4-507f-4d55-998f-f33eabf6369f)

Reason:
Next-Auth for instance requires a NEXT_AUTH_URL to be set. This value will redirect users after OAuth to the domain. When deploying a preview environment this won't work, because it would redirect to whatever value is set in the env vars. Passing the {{domain}} to an environment variable would resolve this issue without having to use any proxy auth server.

**Comment by andrasbacsai** [comment]
There are a few predefined env variables, that you can pass to during build and runtime.

[Here](https://coolify.io/docs/knowledge-base/environment-variables#predefined-variables) you can read more.

Let me know if this does not solve your issue (and reopen this discussion).
